src/record/mod.rs

```diff
@@ -1,10 +1,14 @@
+pub(crate) mod partially_serialized;
+
```

Comment:
I think it is better to place all content to `record.rs` module and in `mod.rs` only re-export all required types